### ISSUE TITLE
Evaluation on test data breaks if data source is a sparse backed object

### ISSUE THREAD

**Issue by Abdul-Moeed** [theislab/sfaira#7]
```
Traceback (most recent call last):
  File "/home/icb/abdul.moeed/sfaira_benchmarks/scripts/grid_searches/train_script_backed_data.py", line 104, in <module>
    specific=True
  File "/mnt/home/icb/abdul.moeed/sfaira/sfaira/train/train_model.py", line 220, in save
    self.save_eval(fn=fn)
  File "/mnt/home/icb/abdul.moeed/sfaira/sfaira/train/train_model.py", line 264, in save_eval
    evaluation_train = self.estimator.evaluate_any(idx=self.estimator.idx_train)
  File "/mnt/home/icb/abdul.moeed/sfaira/sfaira/estimators/keras.py", line 707, in evaluate_any
    x=x, y=y
  File "/home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/keras/engine/training.py", line 108, in _method_wrapper
    return method(self, *args, **kwargs)
  File "/home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/keras/engine/training.py", line 1379, in evaluate
    tmp_logs = test_function(iterator)
  File "/home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/eager/def_function.py", line 780, in __call__
    result = self._call(*args, **kwds)
  File "/home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/eager/def_function.py", line 814, in _call
    results = self._stateful_fn(*args, **kwds)
  File "/home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 2828, in __call__
    graph_function, args, kwargs = self._maybe_define_function(args, kwargs)
  File "/home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 3210, in _maybe_define_function
    return self._define_function_with_shape_relaxation(args, kwargs)
  File "/home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 3142, in _define_function_with_shape_relaxation
    args, kwargs, override_flat_arg_shapes=relaxed_arg_shapes)
  File "/home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 3075, in _create_graph_function
    capture_by_value=self._capture_by_value),
  File "/home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/framework/func_graph.py", line 986, in func_graph_from_py_func
    func_outputs = python_func(*func_args, **func_kwargs)
  File "/home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/eager/def_function.py", line 600, in wrapped_fn
    return weak_wrapped_fn().__wrapped__(*args, **kwds)
  File "/home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/framework/func_graph.py", line 973, in wrapper
    raise e.ag_error_metadata.to_exception(e)
TypeError: in user code:

    /home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/keras/engine/training.py:1224 test_function  *
        return step_function(self, iterator)
    /mnt/home/icb/abdul.moeed/sfaira/sfaira/models/pp_layer.py:14 call  *
        sf = tf.reduce_sum(inputs, axis=1, keepdims=True) / 10000.
    /home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/util/dispatch.py:201 wrapper  **
        return target(*args, **kwargs)
    /home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/ops/math_ops.py:1984 reduce_sum
        _ReductionDims(input_tensor, axis))
    /home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/ops/math_ops.py:1995 reduce_sum_with_dims
        gen_math_ops._sum(input_tensor, dims, keepdims, name=name))
    /home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/ops/gen_math_ops.py:10537 _sum
        name=name)
    /home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/framework/op_def_library.py:473 _apply_op_helper
        raise err
    /home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/framework/op_def_library.py:470 _apply_op_helper
        preferred_dtype=default_dtype)
    /home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/framework/ops.py:1499 convert_to_tensor
        ret = conversion_func(value, dtype=dtype, name=name, as_ref=as_ref)
    /home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/framework/constant_op.py:338 _constant_tensor_conversion_function
        return constant(v, dtype=dtype, name=name)
    /home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/framework/constant_op.py:264 constant
        allow_broadcast=True)
    /home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/framework/constant_op.py:282 _constant_impl
        allow_broadcast=allow_broadcast))
    /home/icb/abdul.moeed/miniconda3/envs/tf23/lib/python3.7/site-packages/tensorflow/python/framework/tensor_util.py:552 make_tensor_proto
        "supported type." % (type(values), values))

    TypeError: Failed to convert object of type <class 'tensorflow.python.framework.sparse_tensor.SparseTensor'> to Tensor. Contents: SparseTensor(indices=Tensor("DeserializeSparse:0", shape
=(None, 2), dtype=int64), values=Tensor("DeserializeSparse:1", shape=(None,), dtype=float32), dense_shape=Tensor("stack:0", shape=(2,), dtype=int64)). Consider casting elements to a supporte
d type.
```

**Comment by Abdul-Moeed** [comment]
This does not happen during training or train_eval because we use a `generator()` to fetch data for the model.

**Comment by davidsebfischer** [comment]
@Abdul-Moeed can we close this?

**Comment by le-ander** [comment]
Closed by #46
